xgcm/grid.py

```diff
@@ -758,6 +766,167 @@ def max(
+                target_dim = target_data.name
+                target = xr.DataArray(
+                    target, dims=[target_dim], coords={target_dim: target}
+                )
+            return target, target_dim, target_data
+
+        def _check_target_alignment(da, target_da):
+            # check alignment for all other axes, to avoid broadcasting enourmous arrays
+            # to do this on the axis level we will simply check if target_data has some coordinates
+            # that are not found in da (excluding all coordinates that belong to the current axis)
+            da_other_dims = set(da.dims) - set(self.coords.values())
+            target_da_other_dims = set(target_da.dims) - set(self.coords.values())
+            if not target_da_other_dims.issubset(da_other_dims):
+                raise ValueError(
+                    f"Found additional dimensions [{target_da_other_dims-da_other_dims}] in `target_data` not found in `da`. This could mean that the target array is not on the same position along other axes. Use grid.interp() to align the arrays."
+                )
+
+        _, dim = self._get_axis_coord(da)
+        if method == "linear":
+            target, target_dim, target_data = _parse_target(
+                target, target_dim, dim, target_data
+            )
+            _check_target_alignment(da, target_data)
+            out = linear_interpolation(
+                da,
+                target_data,
+                target,
+                dim,
+                dim,  # in this case the dimension of phi and theta are the same
+                target_dim,
+                mask_edges=mask_edges,
+            )
+        elif method == "conservative":
+            # the conservative method requires `target_data` to be on the `outer` coordinate.
+            # If that is not the case (a very common use case like transformation on any tracer),
+            # we need to infer the boundary values (using the interp logic)
+            # for this method we need the `outer` position. Error out if its not there.
+            try:
+                target_data_dim = self.coords["outer"]
+            except KeyError:
+                raise RuntimeError(
+                    "In order to use the method `conservative` the grid object needs to have `outer` coordinates."
+                )
+
+            target, target_dim, target_data = _parse_target(
+                target, target_dim, target_data_dim, target_data
+            )
+
+            _check_target_alignment(da, target_data)
+
+            # check on which coordinate `target_data` is, and interpolate if needed
+            if target_data_dim not in target_data.dims:
+                warnings.warn(
+                    "The `target data` input is not located on the cell bounds. This method will continue with linear interpolation with repeated boundary values. For most accurate results provide values on cell bounds.",
```

Comment:
```suggestion
                    "`target_data` is not located on the `outer` Axis position. Data will be interpolated.",
```

Let's use language consistent with our documentation.

Reply:
Great